benchmark/partial_tweets/rapidjson_sax.h

```diff
@@ -0,0 +1,165 @@
+            found_rt = (1<<14),
+            found_fav = (1<<15)
+        };
+        int values = state::key_date;
+        bool userobject_id = false; // If in a user object (to find user.id)
+        bool userobject_screen_name = false;    // If in a user object (to find user.screen_name)
+        bool inretweet = false; // If in a retweet (all keys irrelevant in retweet object)
+        // Fields to store partial tweet info
+        uint64_t user_id;
+        uint64_t id;
+        uint64_t rt;
+        uint64_t fav;
+        uint64_t reply_status;
+        char* screen_name;
+        char* date;
+        char* text;
+        std::vector<tweet<std::string_view>>& result;
+
+        Handler(std::vector<tweet<std::string_view>> &r) : result(r) { }
+
+        bool Key(const char* key, SizeType length, bool copy) {
+            if (!inretweet) {   // If not in a retweet object, find relevant keys
+                if ((length == 16) && (memcmp(key,"retweeted_status",16) == 0)) { inretweet = true; }   // Check if entering retweet
+                else if ((length == 8) && (memcmp(key,"metadata",8) == 0)) { values = 0; }  // Reset
+                // Check if key has been found and if key matches a valid key
+                else if (!(values &(found_date)) && (length == 10) && (memcmp(key,"created_at",10) == 0)) { values |= (key_date); }
+                // Must also check if not in a user object
+                else if (!(values &(found_id)) && !userobject_id && (length == 2) && (memcmp(key,"id",2) == 0)) { values |= (key_id); }
+                else if (!(values &(found_text)) && (length == 4) && (memcmp(key,"text",4) == 0)) { values |= (key_text); }
+                else if (!(values &(found_reply)) && (length == 21) && (memcmp(key,"in_reply_to_status_id",21) == 0)) { values |= (key_reply); }
+                // Check if entering user object
+                else if ((length == 4) && (memcmp(key,"user",4) == 0)) { userobject_id = userobject_screen_name = true; }
+                // Must also check if in a user object
+                else if (!(values &(found_userid)) && userobject_id && (length == 2) && (memcmp(key,"id",2) == 0)) { values |= (key_userid); }
+                // Must also check if in a user object
+                else if (!(values &(found_screenname)) && userobject_screen_name && (length == 11) && (memcmp(key,"screen_name",11) == 0)) { values |= (key_screenname); }
+                else if (!(values &(found_rt)) && (length == 13) && (memcmp(key,"retweet_count",13) == 0)) { values |= (key_rt); }
+                else if (!(values &(found_fav)) && (length == 14) && (memcmp(key,"favorite_count",14) == 0)) { values |= (key_fav); }
+            }
+            else if ((length == 9) && (memcmp(key,"retweeted",9) == 0)) { inretweet = false; }  // Check if end of retweet
+            return true;
+        }
+        bool Uint(unsigned i) {
+            if ((values &(key_userid)) && !(values &(found_userid))) {    // user.id
```

Comment:
Throughout, it would probably be cleaner to write `values & key_userid` (and similarly elsewhere). The parentheses are unneeded and probably harm comprehension.